src/Shared/Microsoft.DotNet.Web.Authentication.Tests/TestAppFactory.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.IO;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Web.Authentication.Tests
+{
+    public class TestAppFactory : WebApplicationFactory<EmptyTestStartup>
+    {
+        private readonly ITestOutputHelper _output;
+        private readonly string _path = Path.GetTempPath();
+        private Action<IServiceCollection> _configureServices;
+        private Action<IApplicationBuilder> _configureBuilder;
+
+        public TestAppFactory(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        public void ConfigureServices(Action<IServiceCollection> configureServices)
+        {
+            _configureServices = configureServices;
+        }
```

Comment:
The fact that ConfigureServices and ConfigureBuilder, cannot be called multiple times, isn't apparent here. Maybe this should be += or Delegate.Combine to support that.

Reply:
Sure, makes sense. :-) Will do! Since it's super useful, and we will undoubtedly use it in more places.